### ISSUE TITLE
pretraining for ML_Decoder’ backbone and openimagev6 dataset

### ISSUE THREAD

**Issue by sorrowyn** [Alibaba-MIIL/ML_Decoder#12]
This is a very Interesting work！
May I ask you two questions?
Q1: You used Open Images pretraining for ML_Decoder’ backbone to get tresnet_l_pretrain_ml_decoder.
parser.add_argument('--model-path', default='https://miil-public-eu.oss-eu-central-1.aliyuncs.com/model-zoo/ML_Decoder/tresnet_l_pretrain_ml_decoder.pth', type=str)
![图片](https://user-images.githubusercontent.com/19569162/151325396-b6af580e-9115-4acc-a81e-d11b2f4b6542.png)

Q2：Will you release the data loading file for openimage?

**Comment by mrT23** [comment]
Q1:
the pretrain weights we uploaded are indeed from open images ( https://miil-public-eu.oss-eu-central-1.aliyuncs.com/model-zoo/ML_Decoder/tresnet_l_pretrain_ml_decoder.pth )
using them, you can fully reproduce the article results.

Q2: we share our open images dataset variant in:
https://github.com/Alibaba-MIIL/PartialLabelingCSL/blob/main/OpenImages.md
i am not sure what you mean by " data loading file ". at this moment, we are not providing a specific dataloader for open images. but you can easily write one yourself.

**Comment by sorrowyn** [comment]
Thank you for your sincere reply!
Now I'm using the penImage dataset that you provided.However, it seems to be a missing JSON file containing labels to the index.
More important. Happy New Year!

**Comment by sorrowyn** [comment]
> Q1: the pretrain weights we uploaded are indeed from open images ( https://miil-public-eu.oss-eu-central-1.aliyuncs.com/model-zoo/ML_Decoder/tresnet_l_pretrain_ml_decoder.pth ) using them, you can fully reproduce the article results.
> 
> Q2: we share our open images dataset variant in: https://github.com/Alibaba-MIIL/PartialLabelingCSL/blob/main/OpenImages.md i am not sure what you mean by " data loading file ". at this moment, we are not providing a specific dataloader for open images. but you can easily write one yourself.

Thank you for your sincere reply!
Now I'm using the penImage dataset that you provided.However, it seems to be a missing JSON file containing labels to the index.
More important. Happy New Year!

**Comment by kprokofi** [comment]
Hello, thanks for your work.
I have another question related to OI.
In your article as well in ASL article you wrote that the number of training images is 9 mln, but in your link for downloading dataset 5662868
Isn't it a mistake?

**Comment by mrT23** [comment]
> > Q1: the pretrain weights we uploaded are indeed from open images ( https://miil-public-eu.oss-eu-central-1.aliyuncs.com/model-zoo/ML_Decoder/tresnet_l_pretrain_ml_decoder.pth ) using them, you can fully reproduce the article results.
> > Q2: we share our open images dataset variant in: https://github.com/Alibaba-MIIL/PartialLabelingCSL/blob/main/OpenImages.md i am not sure what you mean by " data loading file ". at this moment, we are not providing a specific dataloader for open images. but you can easily write one yourself.
> 
> Thank you for your sincere reply! Now I'm using the penImage dataset that you provided.However, it seems to be a missing JSON file containing labels to the index. More important. Happy New Year!

1) Notice that for pre-training or score calculation, you don't need the actual class names, mid description is sufficient
2) I think the file: https://miil-public-eu.oss-eu-central-1.aliyuncs.com/public/Open_Images_V6/mid_to_classes.pth (which appears in https://github.com/Alibaba-MIIL/PartialLabelingCSL/blob/main/OpenImages.md) contains the conversion between mid to class names

**Comment by mrT23** [comment]
> 5662868

The direct flicker links for downloading OpenImages dataset are becoming invalid over the years. In addition, OpenImages also contains different versions (V3,V4,V6).

We downloaded the dataset ~1.5 years ago, and these are the images that were available then. Now there are less. we are the first to try sharing and standardizing the dataset, instead of letting people use fewer and fewer images.
